components/builder-api-client/src/lib.rs

```diff
@@ -931,6 +931,30 @@ impl Client {
         }
     }
 
+    /// Delete a package from Builder
+    ///
+    /// # Failures
+    ///
+    /// * Remote Builder is not available
+    /// * If package does not exist in Builder
+    /// * If the package does not qualify for deletion
+    /// * Authorization token was not set on client
+    pub fn delete_package(&self, ident: &PackageIdent, token: &str) -> Result<()> {
+        if !ident.fully_qualified() {
+            return Err(Error::IdentNotFullyQualified);
+        }
```

Comment:
This check seems superfluous. You can't get to this point unless the package is fully-qualified due to [the `valid_fully_qualified_ident` validator specified in the clap config](https://github.com/habitat-sh/habitat/pull/6350/files#diff-ee10e75b553edc813cf07f6a5036374cR536), and even if it did, builder would reject it.

Reply:
Good pt - removing